src/otx/v2/adapters/torch/mmengine/mmdet/engine.py

```diff
@@ -0,0 +1,214 @@
+                        scheduler.begin = scheduler.end
+                if hasattr(scheduler, "begin") and scheduler.begin > max_epochs:
+                    scheduler.begin = max_epochs - 1
+        return config, update_check
+
+    def predict(
+        self,
+        model: torch.nn.Module | (dict | str) | None = None,
+        img: str | (np.ndarray | list) | None = None,
+        checkpoint: str | Path | None = None,
+        pipeline: dict | list | None = None,
+        device: str | (torch.device | None) = None,
+        task: str | None = None,  # noqa: ARG002
+        batch_size: int = 1,
+        **kwargs,
+    ) -> list[dict]:
+        """Runs inference on the given input image(s) using the specified model and checkpoint.
+
+        Args:
+            model (Optional[Union[torch.nn.Module, Dict, str]], optional): The model to use for inference. Can be a
+                PyTorch module, a dictionary containing the model configuration, or a string representing the path to
+                the model checkpoint file. Defaults to None.
+            img (Optional[Union[str, np.ndarray, list]], optional): The input image(s) to run inference on. Can be a
+                string representing the path to the image file, a NumPy array containing the image data, or a list of
+                NumPy arrays containing multiple images. Defaults to None.
+            checkpoint (Optional[Union[str, Path]], optional): The path to the checkpoint file to use for inference.
+                Defaults to None.
+            pipeline (Optional[Union[Dict, List]], optional): The data pipeline to use for inference. Can be a
+                dictionary containing the pipeline configuration, or a list of dictionaries containing multiple
+                pipeline configurations. Defaults to None.
+            device (Union[str, torch.device, None], optional): The device to use for inference. Can be a string
+                representing the device name (e.g. 'cpu' or 'cuda'), a PyTorch device object, or None to use the
+                default device. Defaults to None.
+            task (Optional[str], optional): The type of task to perform. Defaults to None.
+            batch_size (int, optional): The batch size to use for inference. Defaults to 1.
+            **kwargs: Additional keyword arguments to pass to the inference function.
+
+        Returns:
+            List[Dict]: A list of dictionaries containing the inference results.
+        """
+        import cv2
+        from mmcv.transforms import Compose
+        from mmdet.apis import DetInferencer, inference_detector
+        from mmengine.model import BaseModel
+
+        # Model config need data_pipeline of test_dataloader
+        # Update pipelines
+        if pipeline is None:
+            pipeline = [
+                {"type": "Resize", "scale": [512, 512]},
+                {"type": "PackDetInputs", "meta_keys": ['img', 'img_id', 'scale', 'img_shape', 'scale_factor']},
+            ]
```

Comment:
Let's avoid hard coding. We could define constants like `DEFAULT_PIELINE` next time.

Reply:
Yes, bringing proper way to load default pipeline is the top priority of next phase refactoring. But leave it as it is right now

Reply:
Sure, thanks.